rust/src/state_machine/phases/mod.rs

```diff
@@ -62,17 +63,61 @@ pub trait Handler {
     fn handle_request(&mut self, req: StateMachineRequest) -> Result<(), PetError>;
 }
 
+/// I/O interfaces.
+#[derive(Debug)]
+pub struct IO {
+    /// The request receiver half.
+    pub(in crate::state_machine) request_rx: RequestReceiver,
+    /// The event publisher.
+    pub(in crate::state_machine) events: EventPublisher,
+}
+
+/// A struct that contains the coordinator state and the I/O interfaces.
+#[derive(Debug)]
+pub struct Shared {
```

Comment:
it's not very clear to me what "shared" means here. perhaps a more descriptive name (and docstring)?

Reply:
```suggestion
/// A struct that contains the coordinator state and the I/O interfaces that is shared and accessible by all PhaseStates.
#[derive(Debug)]
pub struct Shared {
```
maybe like this?

Reply:
Yes, I think it’s better